src/main/java/com/google/sps/data/JourneyHandler.java

```diff
@@ -26,19 +27,29 @@
  * Class that stores and creates journeys for delivery slots.
  */
 public class JourneyHandler {
+  private PathFinder pathFinder;
+
+  public JourneyHandler(PathFinder pathFinder) {
+    this.pathFinder = pathFinder;
+  }
   /**
    * Finds a set of orders and an optimal journey for deliverySlot.
    */
-  public void processDeliveryRequest(DeliverySlot deliverySlot) throws ApiException, IOException, InterruptedException {
-    DeliverySystem deliverySystem = new DeliverySystem();
-    ArrayList<Point> orderedWaypoints = deliverySystem.findOptimalDeliveryJourneyForDeliverySlot(deliverySlot);
-    addDeliveryJourney(orderedWaypoints, deliverySlot);
+  public void processDeliveryRequest(DeliverySlot deliverySlot) throws ApiException, BadRequestException, DataNotFoundException, EntityNotFoundException, IOException, InterruptedException {
```

Comment:
This should have tests.

I guess the test will return unimplemented for now and you can adjust the test in a future PR where you `implementaddDeliveryJourney`.

In general in the future, try to avoid partly implementing something (e.g. this method does 4 things and then returns unimplemented). Either you defer implementation completely, or you implement something completely.

Reply:
I changed the method to only throw `UnsupportedOperationException`.